pkg/controller/v1alpha1/llmisvc/router.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2025 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package llmisvc
+
+import (
+	"cmp"
+	"context"
+	"fmt"
+	"slices"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/utils/ptr"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/kmeta"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+)
+
+func (r *LLMISVCReconciler) reconcileRouter(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) error {
+	logger := log.FromContext(ctx).WithName("reconcileRouter")
+	ctx = log.IntoContext(ctx, logger)
+
+	logger.Info("Reconciling Router")
+
+	if err := r.reconcileScheduler(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("SchedulerReconcileError", "Failed to reconcile scheduler: %v", err.Error())
+		return fmt.Errorf("failed to reconcile scheduler: %w", err)
+	}
+
+	// We do not support Gateway's spec, when creating HTTPRoutes either the default gateway or those provided
+	// as refs are attached to reconciled routes
+	if err := r.reconcileHTTPRoutes(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("HTTPRouteReconcileError", "Failed to reconcile HTTPRoute: %v", err.Error())
+		return fmt.Errorf("failed to reconcile HTTP routes: %w", err)
+	}
+
+	llmSvc.MarkRouterReady()
+
+	return nil
+}
+
+func (r *LLMISVCReconciler) reconcileHTTPRoutes(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) error {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling HTTPRoute")
+
+	expectedHTTPRoute := r.expectedHTTPRoute(ctx, llmSvc)
+
+	// TODO should we remove "llmSvc.Spec.Router.Route.HTTP == nil" from the condition below so that a non nil Route meeans "all type of routes are enabled"?
```

Comment:
```suggestion
	// TODO should we remove "llmSvc.Spec.Router.Route.HTTP == nil" from the condition below so that a non nil Route means "all type of routes are enabled"?
```